crates/ruff/src/diagnostics.rs

```diff
@@ -377,13 +376,28 @@ pub(crate) fn lint_stdin(
     noqa: flags::Noqa,
     fix_mode: flags::FixMode,
 ) -> Result<Diagnostics> {
-    // TODO(charlie): Support `pyproject.toml`.
     let source_type = match path.and_then(|path| settings.linter.extension.get(path)) {
         None => match path.map(SourceType::from).unwrap_or_default() {
             SourceType::Python(source_type) => source_type,
-            SourceType::Toml(_) => {
-                return Ok(Diagnostics::default());
+
+            SourceType::Toml(source_type) if source_type.is_pyproject() => {
+                let path = path.unwrap();
```

Comment:
We should only call `lint_pyproject_toml` if the user has any pyproject.toml related rule enabled, similar to 

https://github.com/astral-sh/ruff/blob/74f64d3f96d76fc252512acf30cf45b252be58f7/crates/ruff/src/diagnostics.rs#L229-L239

Could we share some of the logic between `lint_path` and `lint_stdin`

Reply:
I think there's definitely an opportunity to share more but I can see how it is out of the scope for this PR